### DISCUSSION TITLE
Prioritisation Dashboard

### DISCUSSION THREAD

**Discussion by jsvgoncalves** [Depart-de-Sentier/sentier.dev#27] [Workshop Item]
**Goal:** Develop a user-friendly dashboard that allows LCA practitioners and researchers to prioritize which processes or materials to focus on, based on their environmental impact and the currently available data quantity/quality.

**Key Questions:**
- **Data Selection:** 
    - What LCA indicators are most relevant for prioritisation (e.g., carbon footprint, water use, toxicity)?
    - Which data quality metrics are we interested in capturing?
- **Visualization:** How can we present this data in a clear and actionable way (e.g., heatmaps, Pareto charts)?
- **Glossary Integration:** How can we leverage the glossary (api.g.sentier.dev) to enrich the dashboard with definitions, classifications, and contextual information?

**Potential Outcomes:**
- An interactive dashboard that helps LCA practitioners and researchers to make informed decisions about resource allocation.

**Comment by romainsacchi** [comment]
Main objective: visualize the health of a subset of the LCA graph, given one or several impact categories.

Health can be measured/analyzed on the basis of:
* Completeness: technological representation, geographical coverage, emission data gaps (against IO analysis), against standard compliance (ESG, GHG, PEF), commodities that miss a glossary match, and temporal coverage (e.g., based on outdated data points).
* Robustness: uncertainty because of small data samples (observations), uncertainty because of outdated data points, missing reviews, outlying flow amounts (compared to external sources), weak links
* Frequency of use: how frequently are processes used in the database? And how much do they contribute to impacts?

Prioritization should probably be a compound/weighting of these three indicators:
- a rarely used process, may not be a top-ranking issue even if incomplete or uncertain
- while a frequently used but incomplete or unreliable process should be prioritized
and should ideally be visualized with an ordered bar chart.

The ranking is specific to an impact category (it differs if considering GWP, freshwater use, etc.).

Plotting is probably an important feature for understanding the performance of a subset of data and communicating the findings and issues to collaborators and reviewers.

For example:
* heatmaps of flows based on the age of data
* heatmaps highlighting reviewed vs. unreviewed data
* heatmaps highlighting flow amounts based on small data samples, pedigree matrix
* visualize the spread of emissions of a process vs. national inventories (e.g., fertilizer emissions modeled and measured)
* histogram of MC samples to see the extent of deviation
* being able to play with processes' parameters with respect to impact results
* box-and-whiskers analysis for the same product across different regions or technologies
* visualize the frequency of processes in a network
* Sankey to communicate on dominant and possibly problematic processes

The dashboard should also be able to compute statistics for a given view/selection: the number of nodes, edges, emission statistics, and geographical coverage (on a map?), to provide an understanding/tangible representation of what the user currently works on.

**Comment by mklarmann** [comment]
Also I suggest calling the "dashboard": workbench (to illustrate the interactivity).
And to follow up, why then not call the "product system model" that drives the infos available in the workbench:
"LCA Analysis Workbench Backend", or something ... "Analysis Workbench Backend" (Material Flows, Calculations Graph, etc.)

**Comment by mklarmann** [comment]
Also when it comes to the actual UI of the workbench, I think it can be as ugly as a jupyter notebook. Maybe even be a jupyter notebook, with a couple of suggested commands to analyze and plot. Maybe even use something like https://github.com/voila-dashboards/voila to get started with, to keep it maximally adaptable.
